pkg/util/crypt/crypt_dev_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package crypt
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/test"
+	"github.com/sylabs/singularity/internal/pkg/util/fs"
+	"github.com/sylabs/singularity/internal/pkg/util/fs/squashfs"
+)
+
+func TestEncrypt(t *testing.T) {
+	test.EnsurePrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	dev := &Device{}
+
+	emptyFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("failed to create temporary file: %s", err)
+	}
+	emptyFile.Close()
+	defer os.Remove(emptyFile.Name())
+
+	// Create a dummy squashfs file
+	dummyDir, err := ioutil.TempDir("", "dummy-fs-")
+	if err != nil {
+		t.Fatalf("failed to create temporary directory: %s", err)
+	}
+	defer os.RemoveAll(dummyDir)
```

Comment:
It wasn't immediately obvious to me that `dummyRootDir` and `dummyRootFile` were cleaned up by this `os.RemoveAll(dummyDir)`. Not sure if comments would help or moving the statement below the other dummy things would help. 

Just an observation. What do you think?

Reply:
I like to keep the defer close to the statement associated to it but I will improve the documentations of the code.